metadata-utils/src/main/java/com/linkedin/metadata/authorization/PoliciesConfig.java

```diff
@@ -255,6 +255,12 @@ public class PoliciesConfig {
       "Manage Glossary Children",
       "The ability to create and delete the children of this entity.");
 
+  // Glossary Node Privileges
+  public static final Privilege MANAGE_GLOSSARY_CHILDREN_RECURSIVELY_PRIVILEGE = Privilege.of(
+    "MANAGE_GLOSSARY_CHILDREN_RECURSIVELY",
+    "Manage Glossary Children Recursively",
+    "The ability to create and delete the children of this entity recusively.");
```

Comment:
I'm mildly concerned about the naming of this privilege as I'm not confident all of our users, especially non-engineering users, will know what "recursively" means.

Maybe we can make this clearer since we'll have two privileges doing similar things but one allows you to only manage direct children while this allows you to manage all children.

So what if we change the description of the first privilege from `"Manage Glossary Children"` to `"Manage Direct Glossary Children"` and this one to be `"Manage All Glossary Children"`?

Then we can say `"The ability to create and delete the direct children of this entity."` for the direct one and `"The ability to create and delete everything underneath this entity."` for the recursive one.

I think we'll also want to add some details for how this works in our privileges docs - I don't think we have anything about MANAGE_GLOSSARY_CHILDREN in it yet - https://datahubproject.io/docs/authorization/policies

Reply:
+1 on the naming suggestions.

One other thing I'm mildly concerned about is the performance of doing this

Reply:
yeah I'd agree that's something to think about at the very least. so basically we'd be adding ~2N number of trips to the db for each level down the glossary they go (one to get the next parent node, another to check permissions on it) unless they have permissions on a lower level parent.

This would happen on any glossary entity page load as well as on any requests to create, edit, or delete any terms or nodes.

Assuming glossaries don't get _ridiculously_ nested, I'm not sure it would be a huge deal to add these extra queries as it's not like we repeat them over and over. We would just need to make sure that we don't query for this privilege on lists of glossary entities so we don't end up doing this recursive request on 1000 terms at once.

Reply:
I have made the naming changes and update the readme with the 2 policies. Agree with @chriscollins3456 , the performance will become a concern if the glossary nodes are deeply nested. It is currently used when retrieving or updating a single glossary node/term. It doesn't get called from any search pages where bulk retrievals are performed.